src/store/token/actions.js

```diff
@@ -107,6 +109,12 @@ export function checkTokenState() {
       const address = await core.token.getAddress(safe.currentAccount);
 
       if (address === ZERO_ADDRESS) {
+        dispatch(
```

Comment:
What about the dispatch on line 129? I think that is the better place to dispatch the error message. We should rather review that current on?
@juanenrisley what do you think?

Reply:
Yeah i agree. Should be better to reuse the dispatch function and just parametrize the data we are going to dispatch rather than calling dispatch multiple times everywhere.

Reply:
Dispatch in line 129 is called once there is an error from catch block. We do not receive an error from core when `address === ZERO_ADDRESS`. So dispatch placed over there will not get triggered.

Reply:
An error is being throwed in line 118 so the catch will be triggered with it anyway: https://github.com/CirclesUBI/circles-myxogastria/blob/c708df4f9e394d34a343e81ac8a95f8b79e60e5f/src/store/token/actions.js#L118

Reply:
true, corrected